### ISSUE TITLE
Key to wake up computer (from sleep) forwarded to swaylock

### ISSUE THREAD

**Issue by NilsIrl** [swaywm/swaylock#132]
version: `1.5`

I'm not sure if this is a regression or is due to different hardware. But now, when I press a key to wakeup my computer from sleep, this key is forwarded to swaylock as well as if I had typed it into swaylock.

**Comment by Druneau** [comment]
Hi I seem to run into this aswell.

I try and remember to clear before entering password, however it's a daily occurence to type in wrong password because of the forwarded key.

**Comment by kennylevinsen** [comment]
This is intentional behavior: swaylock has focus, so it receives all keypresses.

What logic would know that a keypress should be ignored? If it's only one, then what about people who keep slapping keys to make sure the keyboard work up and the laptop unsuspended *and* woke swayidle? It's important that no real keypresses are dropped.

In the meantime, use a modifier key or other key without consequence to swaylock input to wake things up without mistyping your password.

**Comment by Druneau** [comment]
Thank you for the clear explanation.  This helps alot.

I will wake up my computer with the first key from my password!

**Comment by emersion** [comment]
Yeah, I don't see any way to ignore such keypresses, at least from the swaylock and Sway side.

**Comment by barrett-ruth** [comment]
@laktak what do you think about resetting the buffer on `<cr>`? Right now I also run into this, but what I don't get (and what other tools like [slock](https://tools.suckless.org/slock/) provide) is why on `<cr>` the buffer _stays_. So, I type a password, it's _wrong_, then I currently _spam backspace_ until the buffer is cleared. Sometimes, this doesn't even work - is there a simpler way to deal with this?

Also, is this project even still active? I'd gladly like to contribute but issues like this make me dubious.

**Comment by kennylevinsen** [comment]
Backspace and delete removes the last character from the password buffer, while escape, ctrl-C and ctrl-U clears the password buffer.

And yes, the project is active.

**Comment by laktak** [comment]
> [@laktak](https://github.com/laktak) what do you think about resetting the buffer on `<cr>`?

Not sure what you mean. It already does that.
